### PR TITLE
[Internal] CTL: Fixes collection creation

### PR DESCRIPTION
**Pull request by ealsur**
Current db and collection creation was using the Throughput config (default 100K) to provision the database instead of the collection.

Originally the workload was based on Java (https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/ctl/AsyncCtlWorkload.java#L302-L315) which uses Database Shared RU.

This PR adds a separate config in case we want to provision Shared DB Throughput.

### CODE REVIEW

File: Microsoft.Azure.Cosmos.Samples/Tools/CTL/Utils.cs [hunk 1]
```diff
@@ -135,28 +135,44 @@ await Utils.ForEachAsync(documentsToCreate, (Dictionary<string, string> doc)
...
+
                 result.CreatedDatabase = true;
-                database = databaseResponse.Database;
+                database = cosmosClient.GetDatabase(config.Database);
             }
 
             Container container;
             try
             {
                 container = await database.GetContainer(config.Collection).ReadContainerAsync();
```
Comment (j82w) [inline Microsoft.Azure.Cosmos.Samples/Tools/CTL/Utils.cs:156]: Why not use CreateContainerIfNotExists?

File: Microsoft.Azure.Cosmos.Samples/Tools/CTL/Scenarios/ChangeFeedPullScenario.cs [hunk 2]
```diff
@@ -86,7 +86,7 @@ internal class ChangeFeedPullScenario : ICTLScenario
                             using (TimerContext timerContext = metrics.Measure.Timer.Time(readLatencyTimer))
                             {
                                 response = await changeFeedPull.ReadNextAsync();
-                                Utils.LogDiagnsotics(
+                                Utils.LogDiagnostics(
```
Comment (j82w) [inline Microsoft.Azure.Cosmos.Samples/Tools/CTL/Scenarios/ChangeFeedPullScenario.cs:89]: Thanks for fixing this.

File: Microsoft.Azure.Cosmos.Samples/Tools/CTL/Utils.cs (hunk 1 above)
Comment (ealsur) [reply Microsoft.Azure.Cosmos.Samples/Tools/CTL/Utils.cs:156]: Because I need to keep track if the resource was created to delete it afterwards

**Review by kirankumarkolli — DISMISSED** [review]

**Review by j82w — APPROVED** [review]

File: Microsoft.Azure.Cosmos.Samples/Tools/CTL/Utils.cs (hunk 1 above)
Comment (j82w) [reply Microsoft.Azure.Cosmos.Samples/Tools/CTL/Utils.cs:149]: You could just use the status code to check if it was created.

**Pull request merged**